src/deepforest/utilities.py

```diff
@@ -410,37 +413,32 @@ def read_file(
     Args:
         input: Path to file, DataFrame, or GeoDataFrame
         root_dir: Root directory for image files
-        image_path: Assign to all rows (use only for single image)
-        label: Assign to all rows (use only for single label)
+        image_path: Assign image_path column to all rows. The full path to the image file.
+        label: Assign label column to all rows.
 
     Returns:
-        GeoDataFrame with geometry column
+        GeoDataFrame with geometry, image_path, and label columns
     """
-    if image_path is not None:
-        warnings.warn(
-            "You have passed an image_path. "
-            "This value will be assigned to every row in the dataframe. "
-            "Only use this if the file contains annotations for a single image.",
-            UserWarning,
-            stacklevel=2,
-        )
-
-    if label is not None:
-        warnings.warn(
-            "You have passed a label. This value will be assigned to every row in the dataframe. "
-            "Only use this if all annotations share the same label.",
-            UserWarning,
-            stacklevel=2,
-        )
-
     # read file
     if isinstance(input, str):
         if input.endswith(".csv"):
             df = pd.read_csv(input)
         elif input.endswith(".json"):
             df = read_coco(input)
         elif input.endswith((".shp", ".gpkg")):
-            df = shapefile_to_annotations(input, root_dir=root_dir)
+            # Use rgb_path for shapefiles that don't have image_path; fall back to image_path if provided.
+            # Check if shapefile has image_path column
+            if "image_path" not in gpd.read_file(input).columns:
+                if image_path is None:
+                    raise ValueError(
+                        "No image_path column found in shapefile, please specify full path to image file in image_path argument: read_file(input=shp_path, image_path='/path/to/image.tif', ...)"
+                    )
+            df = shapefile_to_annotations(
+                input,
+                rgb=image_path,
+                root_dir=root_dir,
+                label=label,
+            )
```

Comment:
This both reads the shapefile twice and is a little confusing to understand. What about:

```python
gpdf = gpd.read_file(input)
if "image_path" not in gpdf.columns:
    if image_path:
        gpdf["image_path"] = image_path
    else:
        raise ValueError(
            "No image_path column found in shapefile, please specify full path to image file in image_path argument: read_file(input=shp_path, image_path='/path/to/image.tif', ...)"
        )
df = shapefile_to_annotations(
    gpdf,
    root_dir=root_dir,
    label=label,
)
```